libglusterfs/src/inode.c

```diff
@@ -1079,7 +1063,10 @@ __inode_link(inode_t *inode, inode_t *parent, const char *name,
             dentry->parent = __inode_ref(parent, false);
             GF_ATOMIC_INC(parent->kids);
             list_add(&dentry->inode_list, &link_inode->dentry_list);
-            link_inode->ns_inode = __inode_ref(parent->ns_inode, false);
+            if (parent->ns_inode)
+                link_inode->ns_inode = __inode_ref(parent->ns_inode, false);
+            else
```

Comment:
This should never happen. All inodes should have a namespace assigned.

Reply:
ACK - removed.